packages/graphql-mini-transforms/src/document.ts

```diff
@@ -28,17 +33,26 @@ export function cleanDocument(
     addTypename(definition);
   }
 
-  const normalizedSource = minifySource(print(document));
+  const documentSource = print(document);
+  const normalizedSource = minifySource(documentSource);
   const normalizedDocument = parse(normalizedSource);
 
   for (const definition of normalizedDocument.definitions) {
     stripLoc(definition);
   }
 
-  // This ID is a hash of the full file contents that are part of the document,
-  // including other documents that are injected in, but excluding any unused
-  // fragments. This is useful for things like persisted queries.
-  const id = createHash('sha256').update(normalizedSource).digest('hex');
+  let id: string;
+
+  if (generateId === undefined) {
+    // This ID is a hash of the full file contents that are part of the document,
+    // including other documents that are injected in, but excluding any unused
+    // fragments. This is useful for things like persisted queries.
+    id = createHash('sha256')
```

Comment:
I liked that the generator was extracted out.

Suggestion here: 

```ts
function defaultGenerateId(normalizedSource: string) {	
  // This ID is a hash of the full file contents that are part of the document,	
  // including other documents that are injected in, but excluding any unused	
  // fragments. This is useful for things like persisted queries.	
  return createHash('sha256')	
    .update(normalizedSource)	
    .digest('hex');	
}
...

// (Don't add those comments, they are only for the review)
// documentSource is not minified
// normalizedSource is already minified
const id = generateId == null ? defaultGenerateId(normalizedSource) : generateId(documentSource);
```

The current commit has an issue because you still call `minifySource` twice 😄 
```

Reply:
> The current commit has an issue because you still call `minifySource` twice 😄

Ha! That's what I get for rushing.

> I liked that the generator was extracted out.

I did too. I don't love the argument being different for the two, since that might cause confusion, if someone looks at `defaultGenerateId` and doesn't realize the argument is slightly different than user-provided functions. However, it's not a major concern, so I'll go ahead and implement your suggestion.